### DISCUSSION TITLE
[Poll] Should we index directories

### DISCUSSION THREAD

**Discussion by marlonmarcello** [wethegit/corgi#40] [Polls]
Main advantages of indexing are:

1. It's an ECMAScript standard
2. Because it's an `index.js` file it doesn't matter if the sources are different extensions, we can change from `.js` to `.jsx` and it would be fine, we can even export from a `.scss` module if needed
3. Easier to update if directories change
4. Stops this uglyness
```jsx
// different types of exports, default and named
import { castToBool } from "@local/utils/utils"
import classnames from "@local/utils/classnames"
import clamp from "@local/utils/clamp"
import { breakpointMap, breakpointOrder, breakpointRanges, layoutSettings } from "@local/utils/layouts"

// duplication of paths
import BodyScripts from "@local/components/app/body-scripts"
import Footer from "@local/components/app/footer/footer"
import Nav from "@local/components/app/nav/nav"
import Picture from "@local/components/elements/picture" // is it a file or an index??
import BackToTop from "@local/components/elements/back-to-top/back-to-top"
```

With named indexing we get:
```jsx
import {
  clamp,
  castToBool,
  classnames,
  breakpointMap,
  breakpointOrder,
  breakpointRanges,
  layoutSettings
} from "@local/utils"

import {
  BodyScripts,
  Footer,
  Nav
  Picture,
  BackToTop
} from "@local/components/app"
```

**Comment by andrewrubin** [comment]
@marlonmarcello do we need a barrel file for this then, which exports _all _ the components? Or will this import style just work out of the box?

**Comment by NicholasLancey** [comment]
I'm all for this, it worked/looked great on the last Corgi project we worked on.

**Comment by marlonmarcello** [comment]
Vote ended.
Result: Yes
We'll be indexing our directories from now on!
